src/utils/index.js

```diff
@@ -970,18 +970,39 @@ export const computeMarketPositions = (
   }, {})
 
 /**
- * @param {Object.<farmhand.toolType, farmhand.toolLevel>} currentToolLevels
+ * @param {Object.<farmhand.toolType, farmhand.toolLevel>} toolLevels
  * @param {farmhand.toolType} toolType
  * @returns {farmhand.state}
  */
-export const unlockTool = (currentToolLevels, toolType) => {
-  if (currentToolLevels[toolType] === toolLevel.UNAVAILABLE) {
-    return Object.assign({}, currentToolLevels, {
-      [toolType]: toolLevel.DEFAULT,
-    })
+export const unlockTool = (state, toolType) => {
+  const { toolLevels } = state
+
+  if (toolLevels[toolType] === toolLevel.UNAVAILABLE) {
+    return {
+      ...state,
+      toolLevels: {
+        ...toolLevels,
+        [toolType]: toolLevel.DEFAULT,
+      },
+    }
   }
 
-  return currentToolLevels
+  return state
+}
+
+/*
+ * @param {farmhand.state} state
+ * @param {farmhand.stageFocusType} stageType
+ * @return {farmhand.state}
+ */
+export const unlockStage = (state, stageType) => {
+  return {
+    ...state,
+    stagesUnlocked: {
+      ...state.stagesUnlocked,
+      [stageType]: true,
+    },
+  }
```

Comment:
These are much better than what was here before! Now they are functionally reducers, so they should probably be moved to individual files in `src/game-logic/reducers/`.

Reply:
ah yeah. i've moved `unlockTool` but with the change to use entitlements it seems that `unlockStage` isn't even needed any more so i've simply removed that one